dpctl/tensor/libtensor/source/copy_as_contig.cpp

```diff
@@ -0,0 +1,775 @@
+//==- copy_ascontig.cpp - Implementation of _tensor_impl module   -*-C++-*-/==//
+//
+//                      Data Parallel Control (dpctl)
+//
+// Copyright 2020-2024 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// This file defines functions of dpctl.tensor._tensor_impl extensions
+//===----------------------------------------------------------------------===//
+
+#include <algorithm>
+#include <numeric>
+#include <utility>
+#include <vector>
```

Comment:
```suggestion
#include <stdexcept>
#include <vector>
```
For `std::runtime_error`.

This change should probably be made throughout `dpctl` (most implementations use this but don't include the header). That can be deferred.